common/k8stest/util_apps.go

```diff
@@ -182,65 +186,129 @@ func (dfa *FioApp) DeployFio(fioArgsSet common.FioAppArgsSet, podPrefix string)
-	if err != nil {
-		return fmt.Errorf("generating fio pod definition %s, %v", dfa.status.podName, err)
-	}
-	if podObj == nil {
-		return fmt.Errorf("failed to generate fio pod definition")
-	}
-	_, err = CreatePod(podObj, common.NSDefault)
-	if err != nil {
-		return fmt.Errorf("creating fio pod %s, %v", dfa.status.podName, err)
-	}
-	// wait for pod to transition to running or complete whichever is first
-	var phase coreV1.PodPhase
-	var podLogSynopsis *common.E2eFioPodLogSynopsis
-	for secs := 0; secs < DefTimeoutSecs; secs++ {
-		phase, podLogSynopsis, err = CheckFioPodCompleted(dfa.status.podName, common.NSDefault)
+			time.Sleep(1 * time.Second)
+		}
+
+		return fmt.Errorf("pod state is %v, %s", phase, podLogSynopsis)
+	} else {
+		labelKey := "e2e-test"
+		labelValue := "fio"
+		labelselector := map[string]string{
+			labelKey: labelValue,
+		}
+		deployment, err := NewDeploymentBuilder().
+			WithName(dfa.DeployName).
+            		WithNamespace(common.NSDefault).
+            		WithLabelsNew(labelselector).
+            		WithSelectorMatchLabelsNew(labelselector).
+            		WithPodTemplateSpecBuilder(
+                		NewPodtemplatespecBuilder().
+                    		WithLabels(labelselector).
+                    		WithContainerBuildersNew(
+                        		NewContainerBuilder().
+                            		WithName(dfa.DeployName).
+                            		WithImage(common.GetFioImage()).
+                            		WithVolumeDeviceOrMount(dfa.status.volName, dfa.VolType).
+                            		WithImagePullPolicy(coreV1.PullAlways).
+                            		WithArgumentsNew(podArgs)).
+                    		WithVolumeBuilders(
+                        		NewVolumeBuilder().
+                            		WithName("ms-volume").
+                            		WithPVCSource(dfa.status.volName),
+                    		),
+            		).Build()
 		if err != nil {
-			return err
+            return fmt.Errorf("failed to create deployment %s definition object in %s namesppace", dfa.DeployName, common.NSDefault)
+        }
+		
+		if dfa.AppNodeName != "" {
+			deployment.Spec.Template.Spec.NodeName = dfa.AppNodeName
 		}
-		switch phase {
-		case coreV1.PodSucceeded:
-			return nil
-		case coreV1.PodRunning:
-			dfa.status.msv, _ = GetMSV(dfa.status.volUuid)
-			logf.Log.Info("PodRunning", "msv", dfa.status.msv)
-			return nil
-		case coreV1.PodFailed:
-			return fmt.Errorf("pod state is %v, %s", phase, podLogSynopsis)
+
+		// Create the deployment
+		err = CreateDeployment(deployment)
+		if err != nil {
+			return fmt.Errorf("creating fio deployment %s, %v", dfa.DeployName, err)
 		}
 
-		time.Sleep(1 * time.Second)
-	}
+		// Wait for pods to be running
+		for i := 0; i < DefTimeoutSecs; i++ {
+			running, err := VerifyDeploymentReadyReplicaCount(dfa.DeployName, common.NSDefault, 1)
+			if err != nil {
+				return err
+			}
+			if running {
+				dfa.status.msv, _ = GetMSV(dfa.status.volUuid)
```

Comment:
don't this outside of for if pod and deployment is ready and verified and also check error

Reply:
Fixed